### PR TITLE
Refinement for GetDataLakePathPropertiesResult and File::Read API.

### PR DESCRIPTION
**Pull request by katmsft**
Fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/1468
Fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/1472

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp [hunk 1]
```diff
@@ -160,6 +162,14 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake { nam
     Azure::Core::Nullable<Core::DateTime> CopyCompletedOn;
     Azure::Core::Nullable<Core::DateTime> ExpiresOn;
     Azure::Core::Nullable<Core::DateTime> LastAccessedOn;
+    bool IsDirectory;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp:165]: compiler warning

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_utilities.hpp [hunk 2]
```diff
@@ -19,4 +19,6 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake { nam
       const std::string& string,
       std::string::const_iterator& cur);
 
+  bool DoesMetadataIncidateIsDirectory(const Storage::Metadata& metadata);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_utilities.hpp:22]:
```suggestion
  bool MetadataIncidatesIsDirectory(const Storage::Metadata& metadata);
```

File: sdk/storage/azure-storage-files-datalake/src/datalake_utilities.cpp [hunk 3]
```diff
@@ -65,4 +66,15 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake { nam
...
+  bool DoesMetadataIncidateIsDirectory(const Storage::Metadata& metadata)
+  {
+    bool result = false;
+    if (metadata.find(DataLakeIsDirectoryKey) != metadata.end())
+    {
+      result = metadata.at(DataLakeIsDirectoryKey) == "true" ? true : false;
+    }
+
+    return result;
+  }
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_utilities.cpp:79]:
```suggestion
  bool DoesMetadataIncidateIsDirectory(const Storage::Metadata& metadata)
  {
        auto ite = metadata.find(DataLakeIsDirectoryKey);
        return ite != metadata.end() && ite->second == "true";
  }
```

File: sdk/storage/azure-storage-files-datalake/src/datalake_file_client.cpp [hunk 4]
```diff
@@ -278,19 +278,22 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
-  Azure::Core::Response<Models::ReadDataLakeFileResult> DataLakeFileClient::Read(
-      const ReadDataLakeFileOptions& options) const
+  Azure::Core::Response<Models::DownloadDataLakeFileResult> DataLakeFileClient::Download(
+      const DownloadDataLakeFileOptions& options) const
   {
     Blobs::DownloadBlobOptions blobOptions;
     blobOptions.Context = options.Context;
     blobOptions.Range = options.Range;
+    // Uncomment me when blob supports range
+    // blobOptions.RangeHashAlgorithm = options.RangeHashAlgorithm;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_file_client.cpp:288]: resolve this after #1521 is merged

**Comment by katmsft** [comment]
Force pushed to rebase to master for blob change essential to this PR.

**Review by Jinming-Hu — APPROVED** [review]
approved without looking

**Pull request merged**